kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -1374,7 +1374,7 @@ private Att addKoreAttributes(Production prod, SetMultimap<KLabel, Rule> functio
           if (prod.items().apply(i) instanceof NonTerminal) {
             format = format.replaceAll("%" + (i+1) + "(?![0-9])", "%" + (nt++));
           } else if (prod.items().apply(i) instanceof Terminal) {
-            format = format.replaceAll("%" + (i+1) + "(?![0-9])", "%c" + ((Terminal)prod.items().apply(i)).value().replace("\\", "\\\\").replace("$", "\\$") + "%r");
+            format = format.replaceAll("%" + (i+1) + "(?![0-9])", "%c" + ((Terminal)prod.items().apply(i)).value().replace("\\", "\\\\").replace("$", "\\$").replace("%", "%%") + "%r");
```

Comment:
This doesn't result in too many `%` being added? It looks something like `%3` could get turned into `%%cSOME_TERMINAL`, where maybe it should be `%cSOME_TERMINAL`.

Also, why use `replace` for the last two instead of `replaceAll`? Just want you to double-check that this is correct.

Reply:
%c and %r are not part of the string passed to `replace`. We use `replace` because `replace` replaces a string everywhere with another string, whereas `replaceAll` replaces a regular expression with a string containing backreferences. We need the regular expressions in one case, but not the other, thus, we use replaceAll only in that one place.

Reply:
Ah sorry, I misread.